src/include/migraphx/matcher.hpp

```diff
@@ -418,34 +418,41 @@ void find_matches_for(source_location location, Mod& mod, instruction_ref ins, M
                 return;
             if(trace > 1 and trace_for)
                 std::cout << "Match: " << matcher_name << std::endl;
-            auto elapsed_time = time<std::chrono::nanoseconds>([&] {
-                auto r = match_instruction(get_module(mod), ins, m.matcher());
-                if(r.result == get_module(mod).end())
+
+            timer match_timer;
+            auto r = match_instruction(get_module(mod), ins, m.matcher());
+            auto match_time = match_timer.record<std::chrono::nanoseconds>();
```

Comment:
Use the `time` function.

Reply:
Also dont use `std::chrono::nanoseconds`, as it uses integral type. Instead use microseconds with floating-point.